packages/chord-detect/index.ts

```diff
@@ -19,31 +19,73 @@ const namedSet = (notes: string[]) => {
   return (chroma: number) => pcToName[chroma];
 };
 
-export function detect(source: string[]): string[] {
+type DetectOptions = {
+  assumePerfectFifth: boolean
+}
+export function detect(source: string[], options: Partial<DetectOptions>): string[] {
   const notes = source.map((n) => note(n).pc).filter((x) => x);
   if (note.length === 0) {
     return [];
   }
 
-  const found: FoundChord[] = findExactMatches(notes, 1);
+  const found: FoundChord[] = findExactMatches(notes, 1, options);
 
   return found
     .filter((chord) => chord.weight)
     .sort((a, b) => b.weight - a.weight)
     .map((chord) => chord.name);
 }
 
-function findExactMatches(notes: string[], weight: number): FoundChord[] {
+
+const BITMASK = {
+  // 3m 000100000000
+  // 3M 000010000000
+  anyThirds: 384,
+  // 5P 000000010000
+  perfectFifth: 16,
+  // 5d 000000100000
+  // 5A 000000001000
+  nonPerfectFifths: 40,
+  anySeventh: 3,
+}
+
+const testChromaNumber = (bitmask: number) => (chromaNumber: number) => Boolean(chromaNumber & bitmask)
+const hasAnyThird = testChromaNumber(BITMASK.anyThirds)
+const hasPerfectFifth = testChromaNumber(BITMASK.perfectFifth)
+const hasAnySeventh = testChromaNumber(BITMASK.anySeventh)
+const hasNonPerfectFifth = testChromaNumber(BITMASK.nonPerfectFifths)
+
+function hasAnyThirdAndPerfectFifthAndAnySeventh(chordType: ChordType) {
+  const chromaNumber = parseInt(chordType.chroma, 2)
+  return hasAnyThird(chromaNumber) && hasPerfectFifth(chromaNumber) && hasAnySeventh(chromaNumber)
+}
+
+function withPerfectFifth(chroma: string): string {
+  const chromaNumber = parseInt(chroma, 2)
+  return hasNonPerfectFifth(chromaNumber)
+  ? chroma 
+  : (chromaNumber | 16).toString(2)
+}
+type FindExactMatchesOptions = {
+  assumePerfectFifth: boolean
+}
+function findExactMatches(notes: string[], weight: number, options: Partial<FindExactMatchesOptions>): FoundChord[] {
   const tonic = notes[0];
   const tonicChroma = note(tonic).chroma;
   const noteName = namedSet(notes);
-  // we need to test all chormas to get the correct baseNote
+  // we need to test all chromas to get the correct baseNote
   const allModes = modes(notes, false);
 
   const found: FoundChord[] = [];
   allModes.forEach((mode, index) => {
+    const assumedMode = withPerfectFifth(mode)
```

Comment:
I guess we don't need to compute this value unless the `if(options.assumePerfectFifth ...` condition is true. 

Also I'd prefer something like `modeWithPerfectFifth` instead of `assumedMode`

Reply:
How about `    const modeWithPerfectFifth = options.assumePerfectFifth && withPerfectFifth(mode)
` then?

Wanted to avoid re-calculating it every time in the filter method.